src/main/java/dev/snowdrop/release/services/AutoUpdateService.java

```diff
@@ -0,0 +1,165 @@
+     * @param qualifier
+     * @param milestone
+     * @return
+     * @throws Throwable
+     */
+    public File updateBuildConfig(File repo, Release release, final String releaseVersion, final String qualifier, final String milestone) {
+        try {
+            final String[] releaseMMFA = releaseVersion.split("\\.");
+            File buildConfigFile = factory.getBuildConfigRelativeTo(repo, releaseVersion);
+            Map<String, String> variableMap = readVariablesFromFile(new FileInputStream(buildConfigFile));
+            variableMap.put("majorVersion", releaseMMFA[0]);
+            variableMap.put("minorVersion", releaseMMFA[1]);
+            variableMap.put("patchVersion", releaseMMFA[2]);
+            variableMap.put("qualifier", qualifier);
+            variableMap.put("milestone", milestone);
+            BuildConfig buildConfigObj = factory.createFromRepo(buildConfigFile);
+            final String[] releaseMMF = releaseVersion.split("\\.");
+            final String releaseMM = releaseMMF[0] + "." + releaseMMF[1];
+            release.getComponents().stream().forEach(component -> {
+//            // TODO: Check if it's a product or a component only template
+//            manageComponentOnly(component);
+                manageProduct(component, variableMap);
+            });
+
+            factory.saveTo(buildConfigObj, generateVariableConfiguration(variableMap), buildConfigFile);
+            return buildConfigFile;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String generateVariableConfiguration(Map<String, String> variableMap) {
+        return variableMap.entrySet()
+            .stream()
+            .map(entry -> "#!" + entry.getKey() + "=" + entry.getValue())
+            .collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    private void manageProduct(Component component, Map<String, String> variableMap) {
+        final Issue issue = issueSvc.getIssue(component);
+        final String description = issue.getDescription();
+        final String versionSection = description.substring(description.lastIndexOf("==="));
+        final String[] artifactArr = versionSection.split("\\n[\\n \\r]*");
+        // Skip the 1st 2 lines
+        for (int arrayPos = 0; arrayPos < ((artifactArr.length - 1) / 5); arrayPos++) {
+            final String gavText = artifactArr[arrayPos * 5 + 1];
+            if (GAV_NAME_REGEX_PATTERN.reset(gavText).matches()) {
+                final String gid = GAV_NAME_REGEX_PATTERN.group(2);
+                final String aid = GAV_NAME_REGEX_PATTERN.group(4);
+                final String varName = WordUtils.capitalizeFully(aid, '-', '.').replaceAll("-", "");
+                final String productNameText = artifactArr[arrayPos * 5 + 1 + 1];
+                final String productVersionText = artifactArr[arrayPos * 5 + 1 + 2];
+                final String productVersion = productVersionText.split(":")[1];
```

Comment:
Since you're using a pattern to match the GAV, why not tweak it so that you can extract the information you need instead of doing sub-matching?

Reply:
I tried matching both the strings with the same regex but i couldn't find a simple way of doing it. Since the texts are independent and unrelated I think we should keep this as is.